packages/angular_devkit/schematics_cli/bin/schematics.ts

```diff
@@ -354,34 +338,48 @@ function getUsage(): string {
 
 /** Parse the command line. */
 const booleanArgs = [
-  'allowPrivate',
   'allow-private',
   'debug',
   'dry-run',
-  'dryRun',
   'force',
   'help',
   'list-schematics',
-  'listSchematics',
   'verbose',
   'interactive',
 ];
 
-function parseArgs(args: string[] | undefined): minimist.ParsedArgs {
-  return minimist(args, {
+/** Parse the command line. */
+function parseArgs(args: string[]): { _: string[] } & Record<string, unknown> {
+  const { $0, _, ...options } = yargsParser(args, {
```

Comment:
```suggestion
  const { _, ...options } = yargsParser(args, {
```
`$0` doesn't appear used.

Reply:
Actually that is why it’s part in the rest operator. To remove it from options object as otherwise the execution of the schematics fail due to unknown option.

Reply:
Actually that is why it’s part in the rest operator. To remove it from options object as otherwise the execution of the schematics fail due to unknown option.

Reply:
From a quick test, it looks like `yargs-parser` doesn't add the `$0` property:
```
node -p "require('yargs-parser')(['--foo'])"
```
Outputs: `{ _: [], foo: true }`